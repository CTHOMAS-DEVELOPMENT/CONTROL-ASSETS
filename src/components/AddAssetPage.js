import React from "react";
import { connect } from "react-redux";
import AssetForm from "./AssetForm";
import { addAsset } from "../actions/assets";
const AddAssetPage = props => (
  <div>
    <AssetForm
      onSubmit={asset => {
        props.dispatch(addAsset(asset));
        props.history.push("/");
      }}
    />
  </div>
);

export default connect()(AddAssetPage);
