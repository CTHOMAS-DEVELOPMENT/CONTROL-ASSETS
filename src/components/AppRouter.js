import React from "react";
import { BrowserRouter, Route, Switch, Link, NavLink } from "react-router-dom";

import AddExpensePage from "./AddExpensePage";
import EditExpensePage from "./EditExpensePage";
import HelpPage from "./HelpPage";
import NotFoundPage from "./NotFoundPage";
const ExpenseDashboardPage = () => (
  <div>This is from my dashboard component</div>
);

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create Expense
    </NavLink>
    <NavLink to="/edit" activeClassName="is-active">
      Edit Expense
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      Help
    </NavLink>
  </header>
);

const routes = (
  <BrowserRouter>
    <div>
      <Header />
      <Switch>
        <Route path="/" component={ExpenseDashboardPage} exact={true} />
        <Route path="/create" component={AddExpensePage} />
        <Route path="/edit" component={EditExpensePage} />
        <Route path="/help" component={HelpPage} />
        <Route component={NotFoundPage} />
      </Switch>
    </div>
  </BrowserRouter>
);

export default routes;
