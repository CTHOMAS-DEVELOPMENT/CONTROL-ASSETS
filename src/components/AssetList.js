import React from "react";
import { connect } from "react-redux";
import ProductListItem from "./ProductListItem";
import selectExpenses from "../selectors/expenses";

export const AssetList = props => (
  <div className="content-container list-header">
    {console.log("token*", props)}
    {props.products.length === 0 ? (
      <p>No Products</p>
    ) : (
      props.products.map(product => {
        //console.log("product?", product);
        return (
          <ProductListItem
            key={product.projectId}
            product={product}
            token={props.token}
          />
        );
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(AssetList);
