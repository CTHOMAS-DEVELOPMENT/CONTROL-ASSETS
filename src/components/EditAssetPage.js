import React from "react";
import { connect } from "react-redux";
import AssetForm from "./AssetForm";
import { editAsset, removeAsset } from "../actions/assets";
//http://167.99.201.85/v1/projects?filter=%7B%20%22where%22%3A%20%7B%20%22projectId%22%3A%20%22abc-estates%22%20%7D%2C%20%20%22order%22%3A%20%22projectId%20ASC%22%2C%20%20%22limit%22%3A%202%2C%20%20%22fields%22%3A%20%7B%22ownerId%22%3A%20false%7D%20%7D&access_token=X6exxmuh3PXCpHwtWJC5jzwu1r67thPzYh80HrispUwwwpEUfnCIwUPyp1fpMDPW
const EditAssetPage = props => {
  console.log("Token id = ", props.match.params.token);
  console.log("EditAssetPage id = ", props.match.params.id);
  const filter = {
    where: {
      projectId: props.match.params.id
    },
    order: "projectId ASC",
    limit: 1,
    fields: { ownerId: false }
  };
  const fetchProduct = () => {
    const url = `http://167.99.201.85/v1/projects?access_token=${
      props.match.params.token
    }&filter=${filter})`;
    fetch(url, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json"
      }
    })
      .then(response => response.json())
      .then(data => {
        /*this.setState({
          username: username,
          password: password,
          id: id,
          ttl: ttl,
          created: created,
          userId: userId,
          products: data
        });*/

        console.log("data", data);
      })
      .catch(err => console.error("Caught error fetching products: ", err));
  };
  fetchProduct();
  return (
    <div>
      <AssetForm
        expense={props.expense}
        onSubmit={expense => {
          //console.log("Expense", expense);
          props.dispatch(editAsset(props.expense.id, expense));
          props.history.push("/");
        }}
      />
      <button
        onClick={e => {
          //console.log(props.expense.id);
          props.dispatch(removeAsset({ id: props.expense.id }));
          props.history.push("/");
        }}
      >
        Remove
      </button>
    </div>
  );
};
const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};
export default connect(mapStateToProps)(EditAssetPage);
