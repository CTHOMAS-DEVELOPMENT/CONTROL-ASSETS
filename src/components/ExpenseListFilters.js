import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import {
  setTextFilter,
  setSortFilter,
  setStartDate,
  setEndDate
} from "../actions/filters";

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = calendarFocused => {
    this.setState(() => ({ calendarFocused }));
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={e => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={e => {
            props.dispatch(setSortFilter(e.target.value));
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={
            typeof this.props.filters.startDate === "object"
              ? this.props.filters.startDate
              : null
          }
          endDate={
            typeof this.props.filters.endDate === "object"
              ? this.props.filters.endDate
              : null
          }
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => {
            return false;
          }}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
