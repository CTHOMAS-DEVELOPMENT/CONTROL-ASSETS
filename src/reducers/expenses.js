const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_EXPENSE":
      const { description, note, createdAt, amount } = action.updates;
      return state.map(expense => {
        if (expense.id === action.id) {
          expense.description = description;
          expense.note = note;
          expense.createdAt = createdAt;
          expense.amount = amount;
        }
        return expense;
      });
    default:
      return state;
  }
};
