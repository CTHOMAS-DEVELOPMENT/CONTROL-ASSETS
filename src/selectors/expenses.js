import moment from "moment";
export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter(expense => {
      const createdAtMoment = moment(expense.createdAt);
      let startDateMatch = true;
      let endDateMatch = true;
      try {
        startDateMatch = startDate
          ? startDate.isSameOrBefore(createdAtMoment, "day")
          : true;
        endDateMatch = endDate
          ? endDate.isSameOrAfter(createdAtMoment, "day")
          : true;
      } catch (error) {
        //console.error(error);
      }
      const lCaseDesc = expense.description.toLowerCase();
      const lCaseSrch = text.toLowerCase();
      const textMatch = lCaseDesc.includes(lCaseSrch) ? true : false;
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createdAt < b.createdAt ? 1 : -1;
      }
      if (sortBy === "amount") {
        return a.amount < b.amount ? 1 : -1;
      }
    });
};
