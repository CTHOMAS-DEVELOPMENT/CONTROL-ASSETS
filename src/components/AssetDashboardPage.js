import React from "react";
import AssetList from "./AssetList";

const AssetDashboardPage = props => (
  <div>
    <AssetList token={props.token} products={props.products} />
  </div>
);

export default AssetDashboardPage;
