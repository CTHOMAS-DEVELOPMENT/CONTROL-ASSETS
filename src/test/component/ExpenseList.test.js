import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/ExpenseList";
import expenses from "../fixtures/expenses";

test("Should render Expense List correctly..", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});
test("Should render no Expense List correctly with no expenses..", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
