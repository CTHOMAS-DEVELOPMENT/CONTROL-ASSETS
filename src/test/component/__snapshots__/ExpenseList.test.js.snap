// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense List correctly.. 1`] = `
<div>
  <ExpenseListItem
    amount={100}
    createdAt={0}
    description="First record"
    id="1"
    key="1"
    note="The first"
  />
  <ExpenseListItem
    amount={150}
    createdAt={-345600000}
    description="Second record"
    id="2"
    key="2"
    note="The second"
  />
  <ExpenseListItem
    amount={50}
    createdAt={345600000}
    description="Third record"
    id="3"
    key="3"
    note="The third"
  />
</div>
`;

exports[`Should render no Expense List correctly with no expenses.. 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
